File: backend/app/Models/Group.php
<?php

namespace App\Models;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Group extends Model
{
    public $timestamps = false;

    protected $fillable = [
        "name"
    ];
    
    public function users() : BelongsToMany
    {
        return $this->belongsToMany(User::class, "member")->as('member')->withPivot('permission');
    }

    public function courses() : BelongsToMany
    {
        return $this->belongsToMany(Course::class, "course_group");
    }

}

File: tests/BolcsessegBarkaAPITests/ApiLibrary/Group.cs
using Newtonsoft.Json;

namespace ApiLibrary;

public class Group
{
    [JsonProperty("id")]
    public int Id { get; set; }
    
    [JsonProperty("name")]
    public string Name { get; set; }
    
    [JsonProperty("users")]
    public List<UserMemberPivot> Users { get; set; }
    
    [JsonProperty("courses")]
    public List<Course> Courses { get; set; }
}